src/swarm/disconnect.js

```diff
@@ -22,11 +26,7 @@ module.exports = (createCommon, options) => {
 
     after(() => common.teardown())
 
-    it('should disconnect from a peer', (done) => {
-      ipfsA.swarm.disconnect(ipfsB.peerId.addresses[0], done)
-    })
-
-    it('should disconnect from a peer (promised)', () => {
+    it('should disconnect from a peer', () => {
       return ipfsA.swarm.disconnect(ipfsB.peerId.addresses[0])
```

Comment:
Should probably assert that the remote peer is no longer in the peer list (and was to start with).